### ISSUE TITLE
Balloon crashes, Resources$NotFoundException and InvocationTargetException

### ISSUE THREAD

**Issue by serdei85** [skydoves/Balloon#57]
- Library Version: v1.1.7
- Affected Device(s): Nexus 5X, Sony Xperia Z5
 
I have a viewPager with 4 pages. It's MVVM architecture and I call balloons from fragments. I created balloons showing playlist/folderlist etc is empty. (I took them in timer to wait 500 msec but it's irrelevant now I think, the problem is the same without timer too).

Automatic dismissing is not working if I'm swiping fast between the 4 pages

![balloon_bug2](https://user-images.githubusercontent.com/61651827/83411949-8cb48580-a419-11ea-85f0-4f572e8749bd.gif)

So I'm swiping from playlist to player page and right after the ballons popup I'm swiping back.

Because of that, I created manual dismissing when the fragment is gone:

```
override fun setMenuVisibility(menuVisible: Boolean) {
        menuVisibilityPlayer = menuVisible

        if (menuVisibilityPlayer && this::binding.isInitialized) {
            createBalloons()
        }else {
            dismissBalloons()
        }
        super.setMenuVisibility(menuVisible)
    }
```

```
    fun dismissBalloons(){
        if (timerBalloon != null){timerBalloon!!.cancel()}
        if (timerBalloon2 != null){timerBalloon2!!.cancel()}

        if (inPlayerEmptyFoldersListBalloonFactory != null) {
            if (inPlayerEmptyFoldersListBalloonFactory!!.isShowing) {
                inPlayerEmptyFoldersListBalloonFactory!!.dismiss()
            }
        }
        if (inPlayerEmptySelectedFoldersListBalloonFactory != null) {
            if (inPlayerEmptySelectedFoldersListBalloonFactory!!.isShowing) {
                inPlayerEmptySelectedFoldersListBalloonFactory!!.dismiss()
            }
        } ...
```

It's seems it's working but very rarely it crashes (I saw on my phone (Xperia Z5) about 2-3 times until a lot of test, but on emulators don't have problems)
I thougth it was fixed with your 1.1.7 version, because I couldn't reproducate it.
But after I made my release, I got crashes in chrashlytics from a Nexus 5x. It comes from my dismissBalloons method:
![image](https://user-images.githubusercontent.com/61651827/83412621-ba4dfe80-a41a-11ea-9ed1-46652cf20ddb.png)

Thank you for helping!

**Comment by skydoves** [comment]
Could you please rebuild using the `setLifecycleOwner` method?
The lifecycleOwner should be `viewlifecycleowner`, not fragment's lifecycle owner.

**Comment by serdei85** [comment]
```
class InPlayerEmptyFoldersListBalloonFactory : Balloon.Factory() {

  override fun create(context: Context, lifecycle: LifecycleOwner?): Balloon {
    val textForm = textForm(context) {
      setTextResource(R.string.emptylistfolder)
      setTextSize(15f)
    }

    return createBalloon(context) {
        setArrowSize(10)
        setArrowPosition(0.5f)
        setCornerRadius(10f)
        setTextForm(textForm)
        setArrowOrientation(ArrowOrientation.BOTTOM)
        setIconDrawableResource(R.drawable.icon_24dp_warning)
        setTextColorResource(R.color.textColor)
        setBackgroundColorResource(R.color.colorBalloon)
        setPadding(6)
        setDismissWhenTouchOutside(true)
        setBalloonAnimation(BalloonAnimation.ELASTIC)
        setLifecycleOwner(lifecycle)
        build()
    }
  }
}
```

Here is my factory. I'm using setLifecycleOwner method. What should I change exactly?

**Comment by skydoves** [comment]
Hmm, I think it is caused because the context is already destroyed in the fragment but tried to create or dismiss the balloon. I recommend to use `viewLifecycleOwner` or `requireActivity()` as a parameter instead of `this`, `requireContext`, `lifecycleOwner`.

**Comment by serdei85** [comment]
Sorry, I newbie in android, I can't figure out where should I modify exactly.

So I have an instance:

`private val inPlayerEmptyFoldersListBalloonFactory by balloon(InPlayerEmptyFoldersListBalloonFactory::class)`

and a showing:

`binding.fixpointForBalloonToFolders.showAlignTop(inPlayerEmptyFoldersListBalloonFactory!!)`

and a factory class:
```
class InPlayerEmptyFoldersListBalloonFactory : Balloon.Factory() {
  override fun create(context: Context, lifecycle: LifecycleOwner?): Balloon {

    return createBalloon(context) {
    ...
        setLifecycleOwner(lifecycle)
        build()
    }
  }
}
```

And I can't modify the constructor or the lifecycle parameter of setLifeCycleOwner.

Thank you for your patience.

**Comment by skydoves** [comment]
Hi, it will be included in the next release: 34d0b5aa
But I think the best way is to dismiss balloon when touched outside.

**Comment by serdei85** [comment]
Hi!

Ok, thank you, I'll check it.

setDismissWhenTouchOutside(true) is in the factory but it's not solve the problem.

As you see on my gif, there will be two "false" balloon from previous page with setDismissWhenTouchOutside(true) and you have to touch outside three times before you can continue to use the app.

Is it not possible that the problem is the animation time? When the balloon start to pop up but it isn't ready yet then it's locking on the screen when you swipe in this exact time.

If the balloon is ready, you can't swipe with the first touch, because the first touch is dismissing the balloon (btw it's embarrassing a little, it will better if the touch would dismiss all balloons and start swiping).

**Comment by skydoves** [comment]
Released a new version `1.2.0`.
Please try using a new version :)

**Comment by serdei85** [comment]
Sorry, I created my own message boxes, which are part of the actual page so it hasn't got swipe or dismiss problems.
